src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -1190,88 +1190,23 @@ namespace Opm {
-                    }
-                    else if (cMode == "GUID"){
+                    prop.handleWELTARG(cmode, newValue, siFactorG, siFactorL, siFactorP);
+                    if (cmode == WellTarget::GUID)
                         well.setGuideRate(currentStep, newValue);
-                    }
-                    else{
-                        throw std::invalid_argument("Invalid keyword (MODE) supplied");
-                    }
 
                     well.setProductionProperties(currentStep, prop);
-                }else{
+                } else {
                     WellInjectionProperties prop = well.getInjectionPropertiesCopy(currentStep);
-                    if (cMode == "BHP"){
-                        prop.BHPLimit = newValue * siFactorP;
-                    }
-                    else if (cMode == "ORAT"){
-                        if(prop.injectorType == WellInjector::TypeEnum::OIL){
-                            prop.surfaceInjectionRate = newValue * siFactorL;
-                        }else{
-                             std::invalid_argument("Well type must be OIL to set the oil rate");
-                        }
-                    }
-                    else if (cMode == "WRAT"){
-                        if(prop.injectorType == WellInjector::TypeEnum::WATER){
-                            prop.surfaceInjectionRate = newValue * siFactorL;
-                        }else{
-                             std::invalid_argument("Well type must be WATER to set the water rate");
-                        }
-                    }
-                    else if (cMode == "GRAT"){
-                        if(prop.injectorType == WellInjector::TypeEnum::GAS){
-                            prop.surfaceInjectionRate = newValue * siFactorG;
-                        }else{
-                            std::invalid_argument("Well type must be GAS to set the gas rate");
-                        }
-                    }
-                    else if (cMode == "THP"){
-                        prop.THPLimit = newValue * siFactorP;
-                    }
-                    else if (cMode == "VFP"){
-                        prop.VFPTableNumber = static_cast<int> (newValue);
-                    }
-                    else if (cMode == "GUID"){
+                    prop.handleWELTARG(cmode, newValue, siFactorG, siFactorL, siFactorP);
+                    if (cmode == WellTarget::GUID)
                         well.setGuideRate(currentStep, newValue);
-                    }
-                    else if (cMode == "RESV"){
-                        prop.reservoirInjectionRate = newValue * siFactorL;
-                    }
-                    else{
-                        throw std::invalid_argument("Invalid keyword (MODE) supplied");
-                    }
 
                     well.setInjectionProperties(currentStep, prop);
                 }
-
-
             }
         }
     }
+        //
```

Comment:
Is there supposed to be text here?

Reply:
No

Reply:
>> Is there supposed to be text here?
>
> No

Okay, then I take it you'll remove the comment sign prior to merging?